src/main/java/org/terasology/core/world/generator/facetProviders/PerlinHillsAndMountainsProvider.java

```diff
@@ -63,9 +63,9 @@ public void process(GeneratingRegion region) {
         SurfaceHumidityFacet humidityData = region.getRegionFacet(SurfaceHumidityFacet.class);
 
         float[] heightData = facet.getInternal();
-        Iterator<BaseVector2i> positionIterator = facet.getRelativeRegion().contents().iterator();
+        Iterator<Vector2ic> positionIterator = facet.getRelativeRegion().iterator();
         for (int i = 0; i < heightData.length; ++i) {
-            BaseVector2i pos = positionIterator.next();
+            Vector2ic pos = positionIterator.next();
```

Comment:
Is this for some reason faster (or otherwise better) than just iterating over the positions in the region and counting the index up?